src/lib.rs

```diff
@@ -513,25 +566,44 @@ fn derive_where_internal(attr: TokenStream, item: TokenStream) -> Result<TokenSt
     } = syn::parse2(item)?;
 
     // Every trait needs a separate implementation.
-    for trait_ in &derive_where.traits {
+    for trait_ in derive_where.traits {
         let body = trait_.generate_body(&ident, &data)?;
-        let trait_ = trait_.type_();
+        let trait_ = trait_.path();
 
         // Build necessary generics to construct the implementation item.
-        let (impl_generics, type_generics, where_clause) = generics.split_for_impl();
-        // TODO: don't convert to `TokenStream`, but actually properly insert bounds.
-        let mut where_clause = where_clause.map(|where_clause| where_clause.into_token_stream());
+        let (impl_generics, type_generics, mut where_clause) = generics.split_for_impl();
```

Comment:
Is there any situation in which these can contain anything?

I don't think the DeriveInput can have a where_clause :thinking:

Reply:
I was surprised myself when I found out that they can:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bc8b601e367637bc848f0c43e100958a

Reply:
You learn something new every day.